### PR TITLE
Improve strings formatting network and offering

### PR DESCRIPTION
**Pull request by Juanpe**
Because #820 was wrongly closed I've created this PR with the same commits based on the `main` branch 😍

### CODE REVIEW

File: Purchases/Logging/Strings/NetworkStrings.swift [hunk 1]
```diff
@@ -17,28 +17,83 @@ import Foundation
...
-    static let retrying_request = "Retrying request %@ %@"
-    static let could_not_find_cached_response =
-        """
-        We were expecting to be able to return a cached response, but we can't find it. Retrying call with a new ETag.
-        """
-    static let could_not_find_cached_response_in_already_retried =
-        """
-        We can't find the cached response, but call has already been retried. Returning result from backend %@.
-        """
+    case api_request_completed(httpMethod: String, path: String, httpCode: Int)
```
Comment (Juanpe) [inline Purchases/Logging/Strings/NetworkStrings.swift:20]: Just to know. I saw some enums with an empty line between the cases and other enums without the empty line. Is there any convection on that?

Comment (aboedo) [reply Purchases/Logging/Strings/NetworkStrings.swift:20]: let's go with no empty lines, it seems to look a little nicer. And the only enum we have in our [SwiftStyleGuide](https://github.com/RevenueCat/purchases-ios/blob/main/SwiftStyleGuide.swift#L87) is using no empty lines

**Review by aboedo — APPROVED** [review]
good to go once the enums are updated to no empty lines! 

thanks so much for your patience with this one, and sorry again for the inconvenience!

**Comment by Juanpe** [comment]
No problem 😉

I've removed the empty lines, so I think it's ready to go 🚀

File: Purchases/Networking/ETagManager.swift [hunk 2]
```diff
@@ -57,8 +57,7 @@ class ETagManager {
                 return storedResponse
             }
             if retried {
-                Logger.warn(String(format: Strings.network.could_not_find_cached_response_in_already_retried,
-                        resultFromBackend.description))
+                Logger.warn(Strings.network.could_not_find_cached_response_in_already_retried(response: resultFromBackend.description))
```
Comment (aboedo) [inline Purchases/Networking/ETagManager.swift:60]: just noticed that this line goes over 120 characters and causes a warning from swiftlint, could you add a line break?

**Review by aboedo — APPROVED** [review]
almost there! just need to update the 120 chars line and we're ready to merge

File: Purchases/Networking/ETagManager.swift (hunk 2 above)
Comment (Juanpe) [reply Purchases/Networking/ETagManager.swift]: Oops 🤦🏼‍♂️

**Comment by aboedo** [comment]
@Juanpe thanks for taking care of this!! Merging now

**Pull request merged**